### ISSUE TITLE
segmentation fault with dual stack config under Ubuntu 16.04

### ISSUE THREAD

**Issue by elsobrino** [acassen/keepalived#1477]
**Describe the bug**
I am trying to get keepalived running with following dual stack configuration. 
```
global_defs {
    notification_email {
        netops@domain.tld
    }
    notification_email_from keepalived@domain.tld
    smtp_server x.x.x.x
    smtp_connect_timeout 30
}

vrrp_instance ipv4 {
    interface bond0
    virtual_router_id 12
    advert_int 1
    smtp_alert
    dont_track_primary
    priority 100
    smtp_alert
    nopreempt

    virtual_ipaddress {
        y.y.y.y
    }
}

vrrp_instance ipv6 {
    interface bond0
    virtual_router_id 12
    advert_int 1
    smtp_alert
    dont_track_primary
    priority 100
    smtp_alert
    nopreempt

    virtual_ipaddress {
        fe80::1111:1
        2001:db8:d::f
    }
}
```

**To Reproduce**
Copy attached keepalived.conf to /etc/keepalived/keepalived.conf and restart keepalived.

**Expected behavior**
Keepalived starts without any problem.

**Keepalived version**
```
Keepalived v1.2.24 (02/14,2019)

Copyright(C) 2001-2019 Alexandre Cassen, <acassen@gmail.com>

Build options:  PIPE2 LIBNL3 RTA_ENCAP RTA_NEWDST RTA_PREF RTA_VIA FRA_OIFNAME FRA_SUPPRESS_PREFIXLEN FRA_SUPPRESS_IFGROUP FRA_TUN_ID RTAX_CC_ALGO RTAX_QUICKACK IPV4_DEVCONF LIBIPTC LIBIPSET LVS LIBIPVS_NETLINK IPVS_SYNCD_ATTRIBUTES IPVS_64BIT_STATS VRRP VRRP_AUTH VRRP_VMAC SOCK_NONBLOCK SOCK_CLOEXEC FIB_ROUTING INET6_ADDR_GEN_MODE SNMP_V3_FOR_V2 SNMP SNMP_KEEPALIVED SNMP_CHECKER SNMP_RFC SNMP_RFCV2 SNMP_RFCV3 SO_MARK
```

**Distro (please complete the following information):**
 - Ubuntu
 - 16.04.6 LTS
 - x86_64

**Details of any containerisation or hosted service (e.g. AWS)**
bare metal

**Configuration file:**
Please see attachment.

**Notify and track scripts**
None

**System Log entries**
```
Jan 14 14:08:34 head1 systemd[1]: Starting Keepalive Daemon (LVS and VRRP)...
Jan 14 14:08:34 head1 Keepalived[1447492]: Starting Keepalived v1.2.24 (02/14,2019)
Jan 14 14:08:34 head1 Keepalived[1447492]: Opening file '/etc/keepalived/keepalived.conf'.
Jan 14 14:08:34 head1 systemd[1]: Started Keepalive Daemon (LVS and VRRP).
Jan 14 14:08:34 head1 Keepalived[1447500]: Starting Healthcheck child process, pid=1447504
Jan 14 14:08:34 head1 Keepalived_healthcheckers[1447504]: Initializing ipvs
Jan 14 14:08:34 head1 Keepalived[1447500]: Starting VRRP child process, pid=1447505
Jan 14 14:08:34 head1 Keepalived_vrrp[1447505]: Registering Kernel netlink reflector
Jan 14 14:08:34 head1 Keepalived_vrrp[1447505]: Registering Kernel netlink command channel
Jan 14 14:08:34 head1 Keepalived_vrrp[1447505]: Registering gratuitous ARP shared channel
Jan 14 14:08:34 head1 Keepalived_vrrp[1447505]: Unable to load ipset library
Jan 14 14:08:34 head1 Keepalived_vrrp[1447505]: Unable to initialise ipsets
Jan 14 14:08:34 head1 Keepalived_vrrp[1447505]: Opening file '/etc/keepalived/keepalived.conf'.
Jan 14 14:08:34 head1 Keepalived_healthcheckers[1447504]: Registering Kernel netlink reflector
Jan 14 14:08:34 head1 Keepalived_healthcheckers[1447504]: Registering Kernel netlink command channel
Jan 14 14:08:34 head1 Keepalived_healthcheckers[1447504]: Opening file '/etc/keepalived/keepalived.conf'.
Jan 14 14:08:34 head1 Keepalived_healthcheckers[1447504]: Using LinkWatch kernel netlink reflector...
Jan 14 14:08:34 head1 Keepalived[1447500]: pid 14475 exited due to segmentation fault (SIGSEGV).
Jan 14 14:08:34 head1 Keepalived[1447500]:   Please report a bug at https://github.com/acassen/keepalived/issues
Jan 14 14:08:34 head1 Keepalived[1447500]:   and include this log from when keepalived started, what happened
Jan 14 14:08:34 head1 Keepalived[1447500]:   immediately before the crash, and your configuration file.
Jan 14 14:08:34 head1 Keepalived[1447500]: VRRP child process(1447505) died: Respawning
Jan 14 14:08:34 head1 Keepalived[1447500]: Starting VRRP child process, pid=1447508
Jan 14 14:08:34 head1 Keepalived_vrrp[1447508]: Registering Kernel netlink reflector
Jan 14 14:08:34 head1 Keepalived_vrrp[1447508]: Registering Kernel netlink command channel
Jan 14 14:08:34 head1 Keepalived_vrrp[1447508]: Registering gratuitous ARP shared channel
Jan 14 14:08:34 head1 Keepalived_vrrp[1447508]: Unable to load ipset library
Jan 14 14:08:34 head1 Keepalived_vrrp[1447508]: Unable to initialise ipsets
Jan 14 14:08:34 head1 Keepalived_vrrp[1447508]: Opening file '/etc/keepalived/keepalived.conf'.
Jan 14 14:08:35 head1 Keepalived[1447500]: pid 14475 exited due to segmentation fault (SIGSEGV).
Jan 14 14:08:35 head1 Keepalived[1447500]:   Please report a bug at https://github.com/acassen/keepalived/issues
Jan 14 14:08:35 head1 Keepalived[1447500]:   and include this log from when keepalived started, what happened
Jan 14 14:08:35 head1 Keepalived[1447500]:   immediately before the crash, and your configuration file.
Jan 14 14:08:35 head1 Keepalived[1447500]: VRRP child process(1447508) died: Respawning
Jan 14 14:08:35 head1 Keepalived[1447500]: Starting VRRP child process, pid=1447510
Jan 14 14:08:35 head1 Keepalived_vrrp[1447510]: Registering Kernel netlink reflector
Jan 14 14:08:35 head1 Keepalived_vrrp[1447510]: Registering Kernel netlink command channel
Jan 14 14:08:35 head1 Keepalived_vrrp[1447510]: Registering gratuitous ARP shared channel
Jan 14 14:08:35 head1 Keepalived_vrrp[1447510]: Unable to load ipset library
Jan 14 14:08:35 head1 Keepalived_vrrp[1447510]: Unable to initialise ipsets
Jan 14 14:08:35 head1 Keepalived_vrrp[1447510]: Opening file '/etc/keepalived/keepalived.conf'.
Jan 14 14:08:35 head1 Keepalived[1447500]: pid 14475 exited due to segmentation fault (SIGSEGV).
Jan 14 14:08:35 head1 Keepalived[1447500]:   Please report a bug at https://github.com/acassen/keepalived/issues
Jan 14 14:08:35 head1 Keepalived[1447500]:   and include this log from when keepalived started, what happened
Jan 14 14:08:35 head1 Keepalived[1447500]:   immediately before the crash, and your configuration file.
Jan 14 14:08:35 head1 Keepalived[1447500]: VRRP child process(1447510) died: Respawning
Jan 14 14:08:35 head1 Keepalived[1447500]: Starting VRRP child process, pid=1447512
[...]
``` 

**Did keepalived coredump?**
No

**Comment by elsobrino** [comment]
Just saw that there are also some entries in dmesg: ```[Tue Jan 14 14:08:32 2020] keepalived[1447554]: segfault at 8 ip 00007f3392fbc0ce sp 00007ffeb3d7a920 error 4 in libip6tc.so.0.1.0[7f3392fba000+6000]
[Tue Jan 14 14:08:32 2020] keepalived[1447556]: segfault at 8 ip 00007f3392fbc0ce sp 00007ffeb3d7a920 error 4 in libip6tc.so.0.1.0[7f3392fba000+6000]
[Tue Jan 14 14:08:33 2020] keepalived[1447558]: segfault at 8 ip 00007f3392fbc0ce sp 00007ffeb3d7a920 error 4 in libip6tc.so.0.1.0[7f3392fba000+6000]
[Tue Jan 14 14:08:33 2020] keepalived[1447560]: segfault at 8 ip 00007f3392fbc0ce sp 00007ffeb3d7a920 error 4 in libip6tc.so.0.1.0[7f3392fba000+6000]```

**Comment by elsobrino** [comment]
Seems to be related to #457

**Comment by elsobrino** [comment]
Loading `ip6_tables` solves it.

**Comment by elsobrino** [comment]
This seems to be necessary, because keepalived creates rules by default. Is it possible do disable this behavior? I don't want keepalived to create any iptables rules. This is partly because we use nftables instead of iptables.

**Comment by pqarmitage** [comment]
@elsobrino To build keepalived without iptables support, run configure with the `--disable-iptables` option, although I don't recall if that option was available as long ago as v1.2.24.

If you upgrade to a recent version of keepalived (v2.0.11 was the first version to support nftables, but v.2.0.19 is recommended), then it supports using nftables. However, since you are not using `no_accept` or using VMACs, then there will be no nftables configuration added by keepalived. Indeed, with current versions of keepalived I don't think with you configuration that any iptables configuration will be added.
